### PR TITLE
Dt/task/swmaas 1533 fix repeating voice instructions 2

### PR DESCRIPTION
**Pull request by dtran-phunware**
###### Context
QA reported voice instructions were repeating for the same step. This PR resolve the issue by ignoring onManueverChanged calls with the same position.

###### Description
* Added logic gate to ignore onManeuverChanged calls with same position as one that preceded it
* Added flags in RoutingActivity and NavOverlayView to determine where a maneuver originated from.
* Added comments for this potentially confusing logic flow

##### Notes
It is possible and valid for multiple instructions to be TTS and displayed on the Pager within rapid succession. As long as the instruction was different form the one that preceded it.

*Screen Record*
https://drive.google.com/file/d/1JPw9EdAo0No5qC7ie2Evk5lLhiYn0Zcq/view?usp=sharing

### CODE REVIEW

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/RoutingActivity.kt [hunk 1]
```diff
@@ -76,7 +75,7 @@ open class RoutingActivity : AppCompatActivity(), OnPhunwareMapReadyCallback,
         Building.OnFloorChangedListener, Navigator.OnManeuverChangedListener,
         LocationManager.LocationListener, RoutingDialogFragment.RoutingDialogListener {
     companion object {
-        private val TAG = RoutingActivity::class.java.simpleName
+        private val TAG = "VoiceRepeatDebug"
```
Comment (avakkachen) [inline Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/RoutingActivity.kt]: Do we want to keep this debug tag?

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/VoicePromptActivity.kt [hunk 2]
```diff
@@ -98,29 +111,18 @@ class VoicePromptActivity : RoutingActivity(), TextToSpeech.OnInitListener {
...
      */
-    override fun onManeuverChanged(navigator: Navigator, position: Int) {
-        super.onManeuverChanged(navigator, position)
-
-        this.currentManeuverPosition = position
-
+    override fun dispatchManeuverChanged(navigator: Navigator, position: Int) {
         // Play the text that is associated with the maneuver position
         if (voiceEnabled) {
+            Log.d("VoiceRepeatDebug", "Voicing TTS for position: $position")
```
[the remark this answers is not in the document]
Comment (npike) [reply Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/VoicePromptActivity.kt]: We can probably remove this log statement too.  I think it was there for debugging issues QA noticed.

**Comment by npike** [comment]
Do we need to update the other activities to implement `dispatchManeuverChanged()` instead of overriding `onManeuverChanged()`?  (like WalkTimeActivity)

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/VoicePromptActivity.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dtran-phunware) [reply Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/VoicePromptActivity.kt]: good catch, yes this was something i should remove

File: Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/RoutingActivity.kt (hunk 1 above)
Comment (dtran-phunware) [reply Samples/kotlin/src/main/java/com/phunware/kotlin/sample/routing/RoutingActivity.kt]: good catch reverting

**Pull request merged**